src/utils/validations/component/validations/with-validation.spec.js

```diff
@@ -0,0 +1,348 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import { withValidation } from '.';
+
+const presErr = new Error('this value is required!');
+const notZeroErr = new Error('this is zero!');
+const asyncErr = new Error('not foo!');
+const failErr = new Error("It's foo!");
+
+const presence = value => new Promise((resolve, reject) => {
+  if (value) {
+    resolve(true);
+  } else {
+    reject(presErr);
+  }
+});
+
+const isNotZero = value => new Promise((resolve, reject) => {
+  if (value !== 0) {
+    resolve(true);
+  } else {
+    reject(notZeroErr);
+  }
+});
+
+const asyncValidation = value => new Promise((resolve, reject) => {
+  setTimeout(() => {
+    if (value === 'foo') {
+      resolve();
+    } else {
+      reject(asyncErr);
+    }
+  }, 200);
+});
+
+const failValidation = val => new Promise((resolve, reject) => {
+  if (val !== 'foo') resolve(true);
+  reject(failErr);
+});
+
+const checkState = (ids, state, res) => {
+  Object.keys(state).forEach((key) => {
+    if (ids.includes(key)) {
+      ids.forEach(id => expect(state[id]).toEqual(res));
+    } else {
+      expect(state[key]).toEqual(!res);
+    }
+  });
+};
```

Comment:
I think that we shouldn't separate `expect` methods from the tests. It decreases code readability.

Reply:
Yeah, I removed it, what I was using the `expect` for was pointless really anyway